storage/containers.go

```diff
@@ -170,7 +170,7 @@ func newContainerStore(dir string) (ContainerStore, error) {
 	if err := os.MkdirAll(dir, 0700); err != nil {
 		return nil, err
 	}
-	lockfile, err := GetLockfile(filepath.Join(dir, "containers.lock"))
+	lockfile, err := GetLockfile(filepath.Join(lockDir, dir, "containers.lock"))
```

Comment:
`dir` is already an absolute path, so this creates a lock file named `/run/storage/locks/var/lib/containers/storage/containers/containers.lock`.  It looks weird, but it's consistent and unique among different graph roots, so I guess it's fine.